resources/codegen_inputs/os2.rs

```diff
@@ -0,0 +1,116 @@
+#![parse_module(read_fonts::tables::os2)]
+
+/// [`OS2`](https://docs.microsoft.com/en-us/typography/opentype/spec/os2)
+table Os2 {
+    #[version]
+    #[compile(self.compute_version())]
+    version: BigEndian<u16>,
+    /// The Average Character Width parameter specifies the arithmetic average
+    /// of the escapement (width) of all non-zero width glyphs in the font.
+    x_avg_char_width: BigEndian<i16>,
+    /// Indicates the visual weight (degree of blackness or thickness of
+    /// strokes) of the characters in the font. Values from 1 to 1000 are valid.
+    us_weight_class: BigEndian<u16>,
+    /// Indicates a relative change from the normal aspect ratio (width to
+    /// height ratio) as specified by a font designer for the glyphs in a font.
+    us_width_class: BigEndian<u16>,
+    /// Indicates font embedding licensing rights for the font.
+    fs_type: BigEndian<u16>,
+    /// The recommended horizontal size in font design units for subscripts for
+    /// this font.
+    y_subscript_x_size: BigEndian<i16>,
+    /// The recommended vertical size in font design units for subscripts for
+    /// this font.
+    y_subscript_y_size: BigEndian<i16>,
+    /// The recommended horizontal offset in font design units for subscripts
+    /// for this font.
+    y_subscript_x_offset: BigEndian<i16>,
+    /// The recommended vertical offset in font design units for subscripts
+    /// for this font.
+    y_subscript_y_offset: BigEndian<i16>,
+    /// The recommended horizontal size in font design units for superscripts
+    /// for this font.
+    y_superscript_x_size: BigEndian<i16>,
+    /// The recommended vertical size in font design units for superscripts
+    /// for this font.
+    y_superscript_y_size: BigEndian<i16>,
+    /// The recommended horizontal offset in font design units for superscripts
+    /// for this font.
+    y_superscript_x_offset: BigEndian<i16>,
+    /// The recommended vertical offset in font design units for superscripts
+    /// for this font.
+    y_superscript_y_offset: BigEndian<i16>,
+    /// Thickness of the strikeout stroke in font design units.
+    y_strikeout_size: BigEndian<i16>,
+    /// The position of the top of the strikeout stroke relative to the
+    /// baseline in font design units.
+    y_strikeout_position: BigEndian<i16>,
+    /// This parameter is a classification of font-family design.
+    s_family_class: BigEndian<i16>,
+    /// Additional specifications are required for PANOSE to classify non-Latin
+    /// character sets.
+    panose_10: BigEndian<u8>,
```

Comment:
This should be a 10 byte array. @cmyr does codegen support something like this?

Reply:
I tried "panose_10: [u8;10]," - that did not work.

Reply:
might be worth trying:
```rust
#[count(10)]
panose: [BigEndian<u8>],
```
No idea if this will work

Reply:
If not, we can create a new Panose struct that holds a [u8; 10] and impl that necessary traits for it

Reply:
IMHO we should support spelling this `panose: [u8, 10]`. No annotation, that's sufficient to know it's an array and it's size. We'd have to teach the codegen the new trick but I believe it achievable based on exploratory hackery in #70.

Reply:
> might be worth trying:
> 
> ```rust
> 
> #[count(10)]
> 
> panose: [BigEndian<u8>],
> 
> ```
> 
> No idea if this will work

this should work, and would be my first thought? Not opposed to fixed-size arrays either, although do we have any idea if we would need them elsewhere?